shlomobot_pytest/utils.py

```diff
@@ -77,18 +89,60 @@ def calculate_total_deducted_score(
     return min(total_points_deducted, max_points_deducted)
 
 
-def import_pyfile(module_name: str) -> ModuleType:
+def import_pyfile(py_filename: str) -> ModuleType:
     """Extracts the module from a given filename"""
-    stripped_module_name = module_name.removesuffix(".py")
+    stripped_module_name = py_filename.removesuffix(".py")
 
     return import_module(stripped_module_name)
 
 
-def get_functions_from_files(file_list: list[str]) -> list[FunctionType]:
+def get_functions_from_files(file_list: list[str]) -> Generator:
```

Comment:
You can be specific in what the generator generates

```suggestion
def get_functions_from_files(file_list: list[str]) -> Generator[tuple[str, list[FunctionType]]]:
```